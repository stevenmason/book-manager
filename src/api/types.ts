import { Book } from "../types";

export interface BooksResponse {
  books: Book[];
  count: number;
}
