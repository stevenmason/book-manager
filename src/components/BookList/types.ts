import { Book } from "../../types";

export interface Props {
  books: Book[];
  isLoading: boolean;
}
