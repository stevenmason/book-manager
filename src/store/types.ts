import { setBooks, setIsLoading, setFilter, setOpenPage } from "./actions";
import getRootReducer from "./reducer";

export type State = ReturnType<ReturnType<typeof getRootReducer>>;

export type Actions =
  | ReturnType<typeof setBooks>
  | ReturnType<typeof setOpenPage>
  | ReturnType<typeof setIsLoading>
  | ReturnType<typeof setFilter>;
