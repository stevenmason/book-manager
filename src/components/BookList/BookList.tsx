import React from "react";
import {
  List,
  ListItem,
  ListItemText,
  CircularProgress,
} from "@material-ui/core";

import { Props } from "./types";
import useStyles from "./styles";

const Body: React.FunctionComponent<Props> = ({ books, isLoading }) => {
  const classes = useStyles();
  return (
    <div className={classes.container}>
      {isLoading && <CircularProgress className={classes.loader} />}
      {books && (
        <List>
          {books.map((book) => (
            <ListItem key={book.id}>
              <ListItemText
                primary={book.book_title}
                secondary={book.book_author}
              />
            </ListItem>
          ))}
        </List>
      )}
    </div>
  );
};

export default Body;
