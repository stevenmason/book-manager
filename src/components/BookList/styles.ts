import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles({
  container: {
    height: "100%",
    overflowY: "auto",
    display: "flex",
  },
  loader: {
    alignSelf: "center",
    margin: "auto",
  },
});

export default useStyles;
