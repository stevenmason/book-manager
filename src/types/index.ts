import { Book, PagedBooks } from "./books";

export type { Book, PagedBooks };
