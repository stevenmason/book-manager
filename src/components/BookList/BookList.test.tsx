import React from "react";
import { render } from "@testing-library/react";

import BookList from "./BookList";
import { Props } from "./types";

const PROPS: Props = {
  books: [
    {
      book_author: ["test"],
      book_pages: 2,
      book_publication_city: "test",
      book_publication_country: "test",
      book_publication_year: 2020,
      book_title: "test",
      id: 1,
    },
    {
      book_author: ["test 2"],
      book_pages: 10,
      book_publication_city: "test 2",
      book_publication_country: "test 2",
      book_publication_year: 2021,
      book_title: "test 2",
      id: 2,
    },
  ],
  isLoading: false,
};

it("renders correctly", () => {
  const { container } = render(<BookList {...PROPS} />);
  expect(container.firstChild).toMatchSnapshot();
});
