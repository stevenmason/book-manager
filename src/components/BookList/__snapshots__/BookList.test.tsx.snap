// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="makeStyles-container-1"
>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          test
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          test
        </p>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          test 2
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          test 2
        </p>
      </div>
    </li>
  </ul>
</div>
`;
